node/src/effect/requests.rs

```diff
@@ -321,33 +321,40 @@ impl Display for StorageRequest {
     }
 }
 
+/// Details of a request for a list of deploys to propose in a new block.
+#[derive(DataSize, Debug)]
+pub struct ListForInclusionRequest {
+    /// The instant for which the deploy is requested.
+    pub(crate) current_instant: Timestamp,
+    /// Set of block hashes pointing to blocks whose deploys should be excluded.
+    pub(crate) past_deploys: HashSet<DeployHash>,
+    /// The hash of the last block that was finalized at the point when the request was made.
+    pub(crate) last_finalized_block: Option<ProtoBlockHash>,
```

Comment:
Is a proto block hash unique?
Let's say blocks 5 and 7 were empty but block 6 contained my deploy; consensus already knows block 7 is finalized — so my deploy won't be in `past_deploys` — but the block proposer still thinks it's at block 5. Will it receive this request and handle it, and add my deploy again?

Reply:
Good point. It's probably not. So we will have to react to finalization only _after_ the linear chain block hash is known... Or maybe identify finalized blocks by heights instead of by hashes? Everything here is purely internal, so that should be sufficient, no?

Reply:
Right, we could just use the block height! :+1: 
That should be much simpler and faster, actually! :tada: